var/spack/repos/builtin/packages/python/package.py

```diff
@@ -67,9 +67,9 @@ class Python(AutotoolsPackage):
 
     extendable = True
 
-    # --enable-shared is known to cause problems for some users on macOS
+    # FYI --enable-shared was known to cause problems for some users on macOS
     # See http://bugs.python.org/issue29846
-    variant('shared', default=sys.platform != 'darwin',
+    variant('shared', default=True,
```

Comment:
Was there any reason for this being static on OS X? I would like to dig into details.

Reply:
From the aforementioned ticket it is rather unclear (https://bugs.python.org/issue29846) however, it is stated that the issue was there just for "some" macos users.

Reply:
FYI: I tested with 2.7.17 and it worked fine

Reply:
You meant spack built 2.7.17 or external system provided one?

Reply:
The system provided one

Reply:
Alex, here we have to see if spack built python+shared runs fine.

Reply:
Any updates here?

Reply:
I will test this and let you know

Reply:
Ok, so it worked with spack built python@3.7.4

Reply:
Thanks!